src/escape/escape.c

```diff
@@ -189,17 +211,46 @@ int main(int argc, char *argv[]) {
 
                         break;
 
-                case ACTION_UNESCAPE:
+                case ACTION_UNESCAPE: {
+                        _cleanup_free_ char *name = NULL;
+
+                        if (arg_template || arg_instance) {
+                                _cleanup_free_ char *template = NULL;
+
+                                r = unit_name_to_instance(*i, &name);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract instance: %m");
+                                        goto finish;
+                                }
+                                if (!*name) {
+                                        log_error("Unit %s is missing the instance name.", *i);
+                                        r = -EINVAL;
+                                        goto finish;
+                                }
+                                r = unit_name_template(*i, &template);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract template: %m");
+                                        goto finish;
+                                }
+                                if (arg_template && strcmp(arg_template, template)) {
+                                        log_error("Unit %s template %s does not match specified template %s.", *i, template, arg_template);
+                                        r = -EINVAL;
+                                        goto finish;
+                                }
+                        } else
+                                name = strdup(*i);
```

Comment:
Missing oom check. It seems it is not necessary to call `strdup()`...

Reply:
If we don’t `strdup`, then cleanup will attempt to free a string from `argv`, no?